nixos/tests/matrix/matrix-alertmanager.nix

```diff
@@ -0,0 +1,190 @@
+import ../make-test-python.nix (
+  { pkgs, ... }:
+  let
+    # Set up SSL certs for Synapse to be happy.
+    runWithOpenSSL =
+      file: cmd:
+      pkgs.runCommand file {
+        buildInputs = [ pkgs.openssl ];
+      } cmd;
+
+    ca_key = runWithOpenSSL "ca-key.pem" "openssl genrsa -out $out 2048";
+    ca_pem = runWithOpenSSL "ca.pem" ''
+      openssl req \
+        -x509 -new -nodes -key ${ca_key} \
+        -days 10000 -out $out -subj "/CN=snakeoil-ca"
+    '';
+    key = runWithOpenSSL "matrix_key.pem" "openssl genrsa -out $out 2048";
+    csr = runWithOpenSSL "matrix.csr" ''
+      openssl req \
+         -new -key ${key} \
+         -out $out -subj "/CN=localhost" \
+    '';
+    cert = runWithOpenSSL "matrix_cert.pem" ''
+      openssl x509 \
+        -req -in ${csr} \
+        -CA ${ca_pem} -CAkey ${ca_key} \
+        -CAcreateserial -out $out \
+        -days 365
+    '';
+  in
+  {
+    name = "matrix-alertmanager";
+    meta.maintainers = with pkgs.lib.maintainers; [ erethon ];
+
+    nodes = {
+      homeserver =
```

Comment:
nit: Is "homeserver" a weird name for this? Would "matrixserver" be less weird?

Reply:
Homeserver, often abbreviated HS, is what matrix calls the server part of the protocol. It's used this way in both Matrix nomenclature and throughout other nix packages/options https://spec.matrix.org/v1.13/#architecture

Reply:
Ah, nevermind, then!